packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -59,6 +73,9 @@ export default function LinkPreview( {
 		! isEmptyURL &&
 		stripHTML( richData?.title || value?.title || displayURL );
 
+	const isUrlRedundant =
+		value?.url && filterTitleForDisplay( displayTitle ) === displayURL;
```

Comment:
I think this might have a couple of issues: 

1. If `value` doesn't have an `url` prop, shouldn't `isUrlRedundant` be `true`? Currently, it will be `false,` unless we invert the `value?.url` logic. 

2. If I understand correctly, the URL will be redundant if any of the two conditions is independently `false`, meaning we likely have to change `&&` to `||`.

Here's how the changed logic would look like:

```suggestion
		! value?.url || filterTitleForDisplay( displayTitle ) === displayURL;
```

Alternatively, feel free to keep the original logic, refraining from inverting it, if you feel that makes it extra confusing or less readable.

Reply:
@tyxla Thanks for your continuous review and valuable feedback on the PR 🙇. I agree and have updated it as per your suggestion.